### ISSUE TITLE
Consider indexing page url

### ISSUE THREAD

**Issue by jahtalab** [elastic/apm-agent-rum-js#775]
The current Transaction.page.url is not indexed and therefore not searchable (it's only used in the UI). Since searching over the page url has been requested before we should consider indexing this field. 

We should, also, clean up the url for indexing. e.g. removing query strings. another possiblity is to add a `fullUrl` field, that contains the url without any change and set the clean up url on `page.url`

**Comment by vigneshshanmugam** [comment]
What about using the request.url which has the full structure of URL defined properly in the spec.

https://github.com/elastic/apm-server/blob/5d93733d78f0da3a5712bcedc21752fa9292d70e/docs/spec/request.json#L51-L96

Not sure if its searchable as well. /cc @elastic/apm-server Please let us know your thoughts.

**Comment by moix** [comment]
Hi, I was looking for something similar and found this ticket. I'm quite new to elastic apm and rum and are investigating now documents structure, I also read about ecs recommendations and was wondering if the url shouldn't be under url? url.registered_domain looks more appropiate? (https://www.elastic.co/guide/en/ecs/current/ecs-url.html) or am I misunderstanding something?

Thanks!

**Comment by axw** [comment]
I tend to agree, @moix. If we record the page URL as `url.full` in Elasticsearch, then it will be searchable (using `url.full.text`) as it's a multi-field.

@vigneshshanmugam In theory you could send `request.url`, but if you send `request.url` you also have to send `request.method`. It's meant for HTTP requests, not browser page loads (which might be stored locally, and have no corresponding HTTP request...)

We can alternatively update the server to store `context.page.url` as `url.full` in Elasticsearch. I suppose we would need to duplicate the URL in `page.url` for backwards compatibility, at least until 8.0. @jahtalab @vigneshshanmugam what do you think?

**Comment by vigneshshanmugam** [comment]
> We can alternatively update the server to store context.page.url as url.full in Elasticsearch. I suppose we would need to duplicate the URL in page.url for backwards compatibility, at least until 8.0.

This solution sound good to me. Also I am in favour of the URL duplication work to be done in the server as it would reduce the payload size in case if we need to duplicate from RUM side.

**Comment by jahtalab** [comment]
Thanks @axw, duplicating the `context.page.url` as `url.full` sounds good to me, the only concern here is that page.url contains all parts of the url (including the query string and fragment) which in practice could be very long. So the server needs to shorten the url, instead of rejecting the payload if the length exceeds the allowed limit.

Also, if we plan to remove `context.page.url` from 8.0 forward then this requires a change in APM UI to use the new field!

**Comment by axw** [comment]
Cool. I'm going to transfer this to apm-server then, and we'll take it from there.
